tests/test_autonomy/test_cli/test_deploy/test_build/test_deployment.py

```diff
@@ -556,39 +433,59 @@ def test_include_acn_and_hardhat_nodes(
                     str(ROOT_DIR),
                     "--open-autonomy-dir",
                     str(ROOT_DIR),
-                    "--use-hardhat",
-                    "--use-acn",
                 )
             )
 
-        build_dir = self.t / DEFAULT_BUILD_FOLDER
-
-        assert result.exit_code == 0, f"{result.stdout_bytes}\n{result.stderr_bytes}"
+        assert result.exit_code == 0, result.output
         assert build_dir.exists()
 
-        docker_compose_file = build_dir / DockerComposeGenerator.output_name
-        with open(docker_compose_file, "r", encoding="utf-8") as fp:
-            docker_compose = yaml.safe_load(fp)
-
-        assert "acn" in docker_compose["services"]
-        assert "hardhat" in docker_compose["services"]
+        self.check_kubernetes_build(
+            build_dir=build_dir,
+        )
 
-    def test_build_dev_failures(
+    def test_kubernetes_build_password(
         self,
     ) -> None:
         """Run tests."""
+        keys_file = Path(ETHEREUM_ENCRYPTED_KEYS)
+        build_dir = self.t / DEFAULT_BUILD_FOLDER
 
         with mock.patch("os.chown"):
             result = self.run_cli(
                 (
-                    str(self.keys_file),
+                    str(keys_file),
                     "--o",
                     str(self.t / DEFAULT_BUILD_FOLDER),
                     "--force",
-                    "--dev",
+                    "--kubernetes",
+                    "--password",
+                    ETHEREUM_ENCRYPTION_PASSWORD,
                     "--local",
                 )
             )
 
-        assert result.exit_code == 1, result.output
-        assert "Please provide proper value for --packages-dir" in result.output
+        assert result.exit_code == 0, result.output
+        assert build_dir.exists()
+
+        kubernetes_config = self.load_kubernetes_config(
+            path=build_dir,
+        )
+        for resource in kubernetes_config:
+            try:
+                container = resource["spec"]["template"]["spec"]["containers"]
+                agent_vars = {f["name"]: f["value"] for f in container[1]["env"]}
+            except (KeyError, IndexError):
+                continue
+
+            assert agent_vars["AEA_PASSWORD"] == ETHEREUM_ENCRYPTION_PASSWORD
```

Comment:
I don't like this `except continue` pattern. More importantly though, right now if you hit the `except` for all `resource` in `kubernetes_config` you will not evaluate `assert agent_vars["AEA_PASSWORD"] == ETHEREUM_ENCRYPTION_PASSWORD`